app/controllers/amendments_controller.rb

```diff
@@ -92,9 +92,7 @@ def organism
   end
 
   def program
-    @program ||= if params[:program_id]
-                   section&.programs&.find(params[:program_id]) || amendment.organism.programs.find(params[:program_id])
-                 end
+    @program ||= (modification_type.use_field?(:organism) ? organism : section)&.programs&.find(params[:program_id]) if params[:program_id]
```

Comment:
could this be a helper to be used for all the changes on this commit?

```ruby
def programs_parent
  if modification_type.use_field?(:organism)
    organism
  else
    section
  end
end
```

Reply:
Ok, i like it, thanks.